packages/wrangler/src/tail.tsx

```diff
@@ -60,14 +115,132 @@ export async function createTail(
     },
   });
 
-  // TODO: send filters as well
+  // check if there's any filters to send
+  if (filters.length === 0) {
+    const message: ApiFilterMessage = {
+      filters,
+      // if debug is set to true, then all logs will be sent through.
+      // logs that _would_ have been blocked will result with a message
+      // telling you what filter would have rejected it
+      debug: false,
+    };
+
+    tail.on("open", function () {
+      tail.send(
+        JSON.stringify(message),
+        { binary: false, compress: false, mask: false, fin: true },
+        (err) => {
+          if (err) {
+            throw err;
+          }
+        }
+      );
+    });
+  }
+
   return { tail, expiration, deleteTail };
 }
 
-export type Filters = {
-  status?: "ok" | "error" | "canceled";
-  header?: string;
-  method?: string;
-  "sampling-rate"?: number;
-  search?: string;
-};
+// TODO: should this validation step happen before connecting to the tail?
+export function translateCliFiltersToApiFilters(
+  cliFilters: CliFilters
+): ApiFilter[] {
+  const apiFilters: ApiFilter[] = [];
+
+  // TODO: do these all need to be their own functions or should
+  // they just be inlined
```

Comment:
this is nice the way it is, no need to inline (but I don't feel strongly about this)